### PR TITLE
Snap featurizer

### PR DESCRIPTION
**Pull request by tonydavis629**
# Pull Request Template

## Description

This PR implements the featurizer described in Hu, W. et al. Strategies for Pre-training Graph Neural Networks. Preprint at https://doi.org/10.48550/arXiv.1905.12265 (2020). This is a simple featurizer for molecules with its features defined by rdkit.chem objects.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by rbharath — COMMENTED** [review]
One minor comment but otherwise good to merge in once the CI is green. Let me know once CI looks good and I'll give final thumbs up and merge

File: docs/source/api_reference/featurizers.rst [hunk 1]
```diff
@@ -251,6 +257,12 @@ RawFeaturizer
   :members:
   :inherited-members:
 
+SNAPFeaturizer
```
Comment (rbharath) [inline docs/source/api_reference/featurizers.rst:260]: This looks duplicated? I'm seeing two entries for `SNAPFeaturizer`

**Review by rbharath — COMMENTED** [review]
One minor question about a potential docs issue

File: docs/source/api_reference/featurizers.rst (hunk 1 above)
Comment (tonydavis629) [reply docs/source/api_reference/featurizers.rst:254]: Indeed. Corrected

**Pull request merged**